### ISSUE TITLE
Add audio to cards

### ISSUE THREAD

**Issue by david-swift** [david-swift/Flashcards#3]
### Describe the solution you'd like

@wada3n had the idea of recording and adding audio to cards (see #1), since this would be great for language learning. We can add a third button to every flashcard's toolbar (next to tags and delete) that opens a popover for recording audio, and then a button on the flashcard next to the star (if enabled, maybe in another corner) for playing the audio.

### Describe alternatives you've considered

-

### Additional context

This problem has been reported by @wada3n in #1.

**Comment by wada3n** [comment]
I agree with your suggested UI for recording and playing audio, They align perfectly with the goal of creating a seamless learning experience.

I'd like to propose an additional feature: separate audio recordings for each side of flashcard.
To implement this, the audio recording popover could include an option to select the side of the flashcard to which the recording is being added.

Furthermore, let’s provide users with a choice between automatic audio playback and manual playback for solution side, and for questions in the study view, via preferences.

Thanks again!

**Comment by david-swift** [comment]
Thanks for the idea, that's exactly how I'm going to implement it! Thank you so much for taking the time for writing those texts! Please don't expect this feature to be available very soon, because it seems to be rather complex and I don't have experience with recording audio. For audio playback, there seems to be a simple widget ([MediaFile](https://docs.gtk.org/gtk4/class.MediaFile.html)), but recording seems to be more complex. [Vocalis](https://gitlab.gnome.org/World/vocalis/-/tree/main/src?ref_type=heads) might be a good reference here. So please don't expect this feature to be implemented very soon!

**Comment by wada3n** [comment]
You’re welcome! 

After giving it some more thought, I actually think we should completely reconsider adding audio recording altogether.. I believe it would be more in line with GNOME’s philosophy to prioritize enabling users to effortlessly add existing audio files.

Fortunately, there are already outstanding apps specifically designed for audio recording, such as Vocalis and Reco. Integrating with such apps would offer a smoother experience for users.
Of course, this is just my suggestion, and I appreciate your broader perspective on the matter.

**Comment by wada3n** [comment]
Hi @david-swift,

I really appreciate your dedication to this project and your detailed response. I understand your point about offering a seamless recording experience within the app.

However, I was thinking about ways to potentially reduce the complexity of development. What if you included an option to move audio files to the app's directory during import? This way, users wouldn't have to use the app for recording, and you wouldn't have to go through the intricate process of implementing that feature. At the same time, users wouldn't need to keep the audio files separately.

Personally, I would greatly appreciate a recording feature, but I also recognize its complexity. I'm excited to see what you decide to do!

Thanks again for everything.

**Comment by david-swift** [comment]
Yes, it probably makes sense to implement the general support for playing audio files first and add the recording feature in a second step. Thank you so much for sharing your ideas!
